### PR TITLE
Support Python 3.9-3.13

### PR DESCRIPTION
**Pull request by arcondello**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
         type: string
 
     docker:
-      - image: circleci/python:3.9
+      - image: cimg/python:3.12
```
Comment (arcondello) [inline .circleci/config.yml:42]: Needed to update these to get a newer docker for circleci. Also 3.9 is getting pretty venerable at this point so no harm using something newer.

File: setup.cfg [hunk 2]
```diff
@@ -15,15 +15,18 @@ project_urls =
...
+    Operating System :: MacOS
     Development Status :: 4 - Beta
     Programming Language :: Python :: 3
     Programming Language :: Python :: 3 :: Only
-    Programming Language :: Python :: 3.8
     Programming Language :: Python :: 3.9
     Programming Language :: Python :: 3.10
     Programming Language :: Python :: 3.11
     Programming Language :: Python :: 3.12
+    Programming Language :: Python :: 3.12
```
Comment (arcondello) [inline setup.cfg:29]: Oops. Will fix.

File: .circleci/config.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ commands:
     parameters:
```
Comment (thisac) [inline .circleci/config.yml:9]: Could also update the macos orb above to @2.5.2, which added support for xcode>15.1. Seems to work fine anyways though.

**Review by thisac — APPROVED** [review]
LGTM overall!

File: setup.cfg [hunk 4]
```diff
@@ -15,15 +15,18 @@ project_urls =
     Source Code = https://github.com/dwavesystems/dwave-samplers/
 classifiers =
     License :: OSI Approved :: Apache Software License
-    Operating System :: OS Independent
+    Operating System :: Microsoft :: Windows
+    Operating System :: POSIX
+    Operating System :: Unix
+    Operating System :: MacOS
     Development Status :: 4 - Beta
```
[the remark this answers is not in the document]
Comment (arcondello) [reply setup.cfg:22]: We could, but IMO that's beyond the scope of this PR.

File: .circleci/config.yml (hunk 3 above)
Comment (arcondello) [reply .circleci/config.yml:9]: I was trying to minimize changes. We could revisit that in the future.

File: .circleci/config.yml [hunk 5]
```diff
@@ -288,7 +288,7 @@ jobs:
           command: |
             python -m venv env
             . env/bin/activate
-            pip install dist/dwave-samplers*.tar.gz
+            pip install dist/dwave*samplers*.tar.gz
```
[the remark this answers is not in the document]
Comment (arcondello) [reply .circleci/config.yml:291]: Yes

File: .circleci/config.yml [hunk 6]
```diff
@@ -297,7 +297,7 @@ workflows:
       - build-and-test-linux: &build
           matrix:
             parameters:
-              python-version: &python-versions [3.8.9, 3.9.4, 3.10.0, 3.11.0, 3.12.0]
+              python-version: &python-versions [3.9.4, 3.10.0, 3.11.0, 3.12.0, 3.13.0]
```
[the remark this answers is not in the document]
Comment (arcondello) [reply .circleci/config.yml:300]: It ends up being a ~=version. So it doesn't matter. I can add a comment.

**Pull request merged**